units/provisioning_create_host.go

```diff
@@ -463,3 +445,53 @@ func (j *createHostJob) isImageBuilt(ctx context.Context) (bool, error) {
 	}
 	return false, nil
 }
+
+// tryHostReplacement attempts to atomically replace the intent host with the
+// real host that was created by this job. If that fails, it checks to see if
+// the real host already exists. If both of those fail, it will attempt to
+// terminate the cloud host.
+func (j *createHostJob) tryHostReplacement(ctx context.Context, cloudMgr cloud.Manager) error {
+	err := host.UnsafeReplace(ctx, j.env, j.HostID, j.host)
+	if err == nil {
+		return nil
+	}
+
+	grip.Warning(message.WrapError(err, message.Fields{
```

Comment:
Should the warning message be sent after we check for the host? It seems if the host was already successfully swapped there is not point of adding a warning message.

Reply:
It's relevant to know when the host swap has failed in provisioning create host, which is the normal way to do the swap (doing it in the agent is the abnormal route). If it silently fails to swap due to this race without any logging, it'll be more difficult to debug. It's also an uncommon occurrence, so it won't log that much compared to all the other noisy logs that jobs have.